### ISSUE TITLE
Images in plugin used with img get marked as errors

### ISSUE THREAD

**Issue by sc0ttkclark** [10up/plugin-check#757]
Related to this sniff: https://github.com/WordPress/plugin-check/blob/02aeb3e0ec5389c5e0d131796a22da7c8f5d5a1b/phpcs-sniffs/PluginCheck/Sniffs/CodeAnalysis/ImageFunctionsSniff.php#L62

> ERROR | PluginCheck.CodeAnalysis.ImageFunctions.NonEnqueuedImage | Images should be added using wp_get_attachment_image() or similar functions

The problem here is that plugins have their own `<img src="<?php echo esc_url( $image_from_plugin) ); ?>" ... />` image tags output. These trigger the `ERROR` which seems like it should be a `WARNING` perhaps?

**Comment by ernilambar** [comment]
Today, I faced same issue with following code:

```php
<div><img src="<?php echo plugins_url( 'img/logo.png', __FILE__ ) ?>" alt="Logo" class="pe-3" /></div>
```

**Comment by swissspidy** [comment]
Sounds reasonable to make it a warning instead. PRs welcome :-)

----

> Today, I faced same issue with following code:
> 
> <div><img src="<?php echo plugins_url( 'img/logo.png', __FILE__ ) ?>" alt="Logo" class="pe-3" /></div>

Aside: you should really specify width & height attributes :)

**Comment by wplaunchify** [comment]
This really should just be fixed as it provides an endless loop of work for anyone who hasn't discovered this "insiders" tip that one should ignore an ERROR when trying to comply with what has now become rather vigorous code review from the team. It's like a stop sign that folks are told "oh, you don't really have to stop at THAT one...cause it's tuesday!"

Please fix, thanks

I solved it for now using this method:

`<a href="https://somedomain.com" target="_blank">
       <img class="plugin_logo" 
            src="<?php echo esc_url(plugins_url('assets/images/plugin-logo.svg', PLUGIN_FILE)); ?>" 
            alt="Plugin Logo">
   </a>`

**Comment by swissspidy** [comment]
Feel free to submit a PR changing the error type to a warning. Note that performance checks are not required for passing the plugin repository submission.

Also, let's please let's try to maintain a nice tone in this discussion.

**Comment by wplaunchify** [comment]
Hi, not sure where any "not nice" tone was implied...sorry if that's what you read. My intent is to always be polite and courteous.

The plugin review team, unfortunately, has not been as generous as of late with their tone, and that's why this has become a bit of a "take it or leave it" type process... literally.

I don't really know how to change the plugin review code, so the best I could offer was how I got the ERROR to no longer appear. Hopefully that will help someone with more skill/interest in fixing the plugin check code to not show information that adds to what has already become rather frustrating for many of us.

**Comment by dejurin** [comment]
### 1. Registering the image upon plugin activation

```PHP
register_activation_hook(__FILE__, function () {
    $image_path = plugin_dir_path(__FILE__) . 'assets/admin/example-image.png'; // Path to the image file in the plugin directory

    // Check if the file exists
    if (!file_exists($image_path)) {
        return;
    }

    // Check if the image is already registered
    $attachment_id = attachment_url_to_postid(plugin_dir_url(__FILE__) . 'assets/admin/example-image.png');
    if ($attachment_id) {
        return;
    }

    // Copy the file to the WordPress uploads directory
    $upload_dir = wp_upload_dir(); // Get the uploads directory
    $upload_file = $upload_dir['path'] . '/' . basename($image_path);
    copy($image_path, $upload_file);

    // Prepare attachment data
    $attachment = [
        'guid'           => $upload_dir['url'] . '/' . basename($image_path),
        'post_mime_type' => mime_content_type($image_path),
        'post_title'     => sanitize_file_name(basename($image_path)),
        'post_content'   => '',
        'post_status'    => 'inherit',
    ];

    // Insert the attachment into the WordPress Media Library
    $attachment_id = wp_insert_attachment($attachment, $upload_file);

    // Generate metadata and update the attachment
    require_once ABSPATH . 'wp-admin/includes/image.php';
    $attach_data = wp_generate_attachment_metadata($attachment_id, $upload_file);
    wp_update_attachment_metadata($attachment_id, $attach_data);
});
```

### 2. Displaying the image in the admin panel

```
<div class="card">
    <p>
        <strong>Thank you for using this plugin!</strong><br>
        Please rate the plugin to help us improve.<br>
        <a href="https://wordpress.org/support/plugin/example-plugin/reviews/#new-post" target="_blank" rel="noopener">
            <?php
            // Try to get the attachment ID of the image
            $attachment_id = attachment_url_to_postid(plugin_dir_url(__FILE__) . 'assets/admin/example-image.png');
            if ($attachment_id) {
                // Use wp_get_attachment_image() for secure output
                echo wp_get_attachment_image($attachment_id, [110, 20], false, [
                    'alt' => 'Example image',
                ]);
            } else {
                // Fallback: use a direct <img> tag if the image is not registered
                echo '<img src="' . esc_url(plugin_dir_url(__FILE__) . 'assets/admin/example-image.png') . '" width="110" height="20" alt="Example image">';
            }
            ?>
        </a>
    </p>
</div>
```
